pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4850,54 +4851,50 @@ cpdef to_offset(freq, bint is_period=False):
 
             tups = zip(split[0::4], split[1::4], split[2::4])
             for n, (sep, stride, name) in enumerate(tups):
-                if not is_period and name.upper() in c_OFFSET_DEPR_FREQSTR:
-                    warnings.warn(
-                        f"\'{name}\' is deprecated and will be removed "
-                        f"in a future version, please use "
-                        f"\'{c_OFFSET_DEPR_FREQSTR.get(name.upper())}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    name = c_OFFSET_DEPR_FREQSTR[name.upper()]
-                if (not is_period and
-                        name != name.upper() and
-                        name.lower() not in {"s", "ms", "us", "ns"} and
-                        name.upper().split("-")[0].endswith(("S", "E"))):
-                    warnings.warn(
-                        f"\'{name}\' is deprecated and will be removed "
-                        f"in a future version, please use "
-                        f"\'{name.upper()}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    name = name.upper()
-                if is_period and name.upper() in c_REVERSE_OFFSET_DEPR_FREQSTR:
-                    if name.upper().startswith("Y"):
+                if not is_period:
+                    if name.upper() in c_OFFSET_RENAMED_FREQSTR:
                         raise ValueError(
-                            f"for Period, please use \'Y{name.upper()[2:]}\' "
-                            f"instead of \'{name}\'"
+                            f"\'{name}\' is no longer supported for offsets. Please "
+                            f"use \'{c_OFFSET_RENAMED_FREQSTR.get(name.upper())}\' "
+                            f"instead."
                         )
+                    if (name.upper() != name and
+                            name.lower() not in {"s", "ms", "us", "ns"} and
+                            name.upper().split("-")[0].endswith(("S", "E"))):
+                        raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
+                if (is_period and
+                        name.upper() in c_OFFSET_TO_PERIOD_FREQSTR and
```

Comment:
`c_OFFSET_TO_PERIOD_FREQSTR` maps offset aliases to period aliases

So if we're already checked `is_period`, it seems strange to check if `name.upper()` is in `c_OFFSET_TO_PERIOD_FREQSTR.keys()`.

should it be `c_OFFSET_TO_PERIOD_FREQSTR.value()`?